apps/extension/src/routes/popup/enter-password.tsx

```diff
@@ -3,36 +3,87 @@ import { useState } from 'react';
 import { PopupPath } from './paths';
 import { usePopupNav } from '../../utils/navigate';
 import { passwordSelector } from '../../state/password';
+import {
+  Button,
+  Card,
+  CardContent,
+  CardDescription,
+  CardHeader,
+  CardTitle,
+  InputProps,
+} from 'ui/components';
+import { FadeTransition, PasswordInput } from '../../components';
 
 export const EnterPassword = () => {
   const navigate = usePopupNav();
   const { isPassword, setPassword } = useStore(passwordSelector);
-  const [inputTx, setInputTxt] = useState('');
+  const [password, setPasswordValue] = useState('');
+  const [matchError, setMatchError] = useState(true);
+
+  const handleUnlock = () => {
+    void (async function () {
+      if (await isPassword(password)) {
+        setPassword(password);
+        navigate(PopupPath.INDEX);
+      } else {
+        setMatchError(false);
+      }
+    })();
+  };
+
+  const handleChangePassword: InputProps['onChange'] = e => {
+    setPasswordValue(e.target.value);
+    setMatchError(true);
+  };
+
+  const gotoRestorePage = () =>
+    void (async function () {
+      await chrome.tabs.create({
+        url: `${chrome.runtime.getURL('page.html')}`,
+      });
+    })();
```

Comment:
You should move business logic to the slice. Also when you do, make sure you add more tests for this functionality.

Reply:
here we have local business logic that is not needed in other places, so I did not move it to slice

Reply:
here we have local business logic that is not needed in other contexts, so I didn't move it in the slice

Reply:
I think it all relates to the password slice and can be moved there